src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -52,6 +52,51 @@ public static StreamWriter AppendText(string path)
             return new StreamWriter(path, append: true);
         }
 
+#if NET6_0_OR_GREATER
+        public static StreamReader OpenText(string path, FileStreamOptions options)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+
+            return new StreamReader(path, Override(options, FileMode.Open));
+        }
+
+        public static StreamWriter CreateText(string path, FileStreamOptions options)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+
+            return new StreamWriter(path, Override(options, FileMode.Create));
+        }
+
+        public static StreamWriter AppendText(string path, FileStreamOptions options)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+
+            return new StreamWriter(path, Override(options, FileMode.Append));
```

Comment:
@stephentoub What should we do in cases where users have provided wrong (or incomplete) information in `FileStreamOptions`. Like for `File.AppendText(path, new FileStreamOptions { Mode = FileMode.Open })` ?

Reply:
More fundamentally,  why are any of these needed? You can do the exact and things with the FileStream ctor directly.  The original issue cites as a reason that folks are misusing the existing APIs when it comes to async, but I don't believe these new overloads (or the ones originally proposed) will help with that.  I don't know why we're adding all this surface area.  A single File.Open(path, options) seems fine,  for the most general case to connect the File static to FileStream with full control.  Everything else her on File seems gratuitous, and problematic for the reason you highlight.

Reply:
@stephentoub you are right, `File.Open` and `FileInfo.Open` is all we need.

@manandre to avoid a merge conflict with #53669 and even more confusion after we have changed the API surface for the n-th time after it got approved I've pushed the neccessary changes to your branch. I hope that you don't mind.